quinn-h3/src/client.rs

```diff
@@ -57,7 +57,9 @@ impl<'a> ClientBuilder<'a> {
             Client {
                 endpoint,
                 settings: self.settings,
-                log: self.log.unwrap_or(Logger::root(slog::Discard, o!())),
+                log: self
```

Comment:
I don't really understand this change.

Reply:
The idea here is that clippy dislikes if you make a function call in an `unwrap_or()` (or several other methods) because the function call gets evaluated whether it's necessary or not. With `unwrap_or_else()` (generally the `_else()` variants), you pass in a closure that only gets called when it's actually necessary, which should be better for performance reasons.

Reply:
Ah, makes much more sense to me now, thanks.